### PR TITLE
Creating a purchase order the product description doesn't respect partner language

### PR DESCRIPTION
**Pull request by agb80**
Mirroring PR odoo/odoo#6892

### CODE REVIEW

File: addons/purchase/purchase.py [hunk 1]
```diff
@@ -1018,7 +1017,11 @@ def onchange_product_id(self, cr, uid, ids, pricelist_id, product_id, qty, uom_i
             # The 'or not uom_id' part of the above condition can be removed in master. See commit message of the rev. introducing this line.
             dummy, name = product_product.name_get(cr, uid, product_id, context=context_partner)[0]
             if product.description_purchase:
-                name += '\n' + product.description_purchase
+                desciption_purchase_label = ir_translation._get_source(
```
Comment (pedrobaeza) [inline addons/purchase/purchase.py]: Why don't you use lang key on context to get the correct translation?

Comment (agb80) [inline addons/purchase/purchase.py]: `context` is holding the lang for current user, but the intention is to use the partner language, and `lang` variable is used to populate `context_partner`

Comment (pedrobaeza) [inline addons/purchase/purchase.py]: Create another context then. You can re-browse with the context you want.

Comment (agb80) [inline addons/purchase/purchase.py]: Sorry @pedrobaeza I don't get what you mean, if you look at [line 1014](https://github.com/OCA/OCB/pull/238/files#diff-fe8587929089e5fdb6c75687011b1c7eR1014) the product object is already browsed using context with partner lang key and is not getting the correct translation, that's why we invoke the actual translation here, something similar to what is being doing [here](https://github.com/OCA/OCB/blob/7.0/addons/account/account_invoice.py#L670)

Comment (pedrobaeza) [inline addons/purchase/purchase.py]: OK, I get it. Please correct the variable name to `description_purchase_label` or even you can just call it `description`

**Comment by agb80** [comment]
@pedrobaeza  sorry for late answer, correction done!

**Comment by StefanRijnhart** [comment]
I cannot reproduce the problem. Using OCB revision 78964636b

**Comment by StefanRijnhart** [comment]
I see on the upstream PR that someone else came to the same conclusion. :-1: for now then.

**Comment by StefanRijnhart** [comment]
Closing after a year without a clear description how to reproduce.

**Pull request closed without merging**